xdsl/dialects/x86/ops.py

```diff
@@ -91,6 +91,25 @@ def parse(cls, parser: Parser) -> Self:
             regions=regions,
         )
 
+    @classmethod
+    def parse_memory_access_offset(cls, parser: Parser) -> Attribute:
+        offset_type = IntegerType(64, Signedness.SIGNED)
+        temp = parse_optional_immediate_value(
+            parser,
+            offset_type,
+        )
+        if temp is None:
+            temp = IntegerAttr(0, offset_type)
+        return temp
+
+    @classmethod
+    def parse_comma_memory_access_offset(cls, parser: Parser) -> Attribute:
+        offset_type = IntegerType(64, Signedness.SIGNED)
+        temp = IntegerAttr(0, offset_type)
+        if parser.parse_optional_punctuation(",") is not None:
+            temp = cls.parse_memory_access_offset(parser)
+        return temp
```

Comment:
Does this work? Without changing the lines below.
```suggestion
    @classmethod
    def parse_memory_access_offset(cls, parser: Parser) -> Attribute | None:
        return parse_optional_immediate_value(
            parser,
            IntegerType(64, Signedness.SIGNED),
        )

    @classmethod
    def parse_comma_memory_access_offset(cls, parser: Parser) -> Attribute | None:
        if parser.parse_optional_punctuation(",") is not None:
            return cls.parse_memory_access_offset(parser)
```